account_product_fiscal_classification/model/product_template.py

```diff
@@ -0,0 +1,101 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Account Product - Fiscal Classification module for Odoo
+#    Copyright (C) 2014-Today GRAP (http://www.grap.coop)
+#    @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from lxml import etree
+
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+from openerp.osv.orm import setup_modifiers
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    # Field Section
+    fiscal_classification_id = fields.Many2one(
+        comodel_name='account.product.fiscal.classification',
+        string='Fiscal Classification',
+        help="Specify the combination of taxes for this product."
+        " This field is required. If you dont find the correct Fiscal"
+        " Classification, Please create a new one or ask to your account"
+        " manager if you don't have the access right.")
+
+    # Overload Section
+    @api.model
+    def create(self, vals):
+        res = super(ProductTemplate, self).create(vals)
+        res.write_taxes_setting(vals)
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        self.write_taxes_setting(vals)
+        return res
+
+    # View Section
+    def fields_view_get(
+            self, cr, uid, view_id=None, view_type='form', context=None,
+            toolbar=False, submenu=False):
+        """Set 'fiscal_classification_id' as required, by UI, to avoid
+        incompatibility with other modules that could have demo data
+        without fiscal_classification_id."""
+        res = super(ProductTemplate, self).fields_view_get(
+            cr, uid, view_id=view_id, view_type=view_type, context=context,
+            toolbar=toolbar, submenu=submenu)
+        if view_type == 'form':
+            doc = etree.XML(res['arch'])
+            nodes = doc.xpath("//field[@name='fiscal_classification_id']")
+            if nodes:
+                nodes[0].set('required', '1')
```

Comment:
Ok thanks.
About the description there is no problem if you don't create, we inherit and add to it.
We use the name for the code, and description to description of classification, maybe we create the code instead of descritption.
For me it's ok now, I also asked @mileo to test this.